sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -91,6 +91,27 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
     }
   }
 
+  private def isMapWithStringKey(e: Expression): Boolean = if (e.resolved) {
+    e.dataType match {
+      case m: MapType => m.keyType.isInstanceOf[StringType]
+      case _ => false
+    }
+  } else {
+    false
+  }
+
+  private def failUnresolvedAttribute(
+      operator: LogicalPlan,
+      a: Attribute,
+      errorClass: String): Nothing = {
+    val unresolvedAttribute = a.sql
+    val candidates = operator.inputSet.toSeq.map(_.qualifiedName)
+    val orderedCandidates = StringUtils.orderStringsBySimilarity(unresolvedAttribute, candidates)
+    a.failAnalysis(
+      errorClass = errorClass,
+      messageParameters = Array(unresolvedAttribute, orderedCandidates.mkString(", ")))
```

Comment:
Both candidates and unresolvedAttribute should be wrapped with toSQLId

Reply:
I see. I will update this one after https://github.com/apache/spark/pull/36891 is merged.